utils/rjson.cc

```diff
@@ -236,6 +239,65 @@ std::string print(const rjson::value& value, size_t max_nested_level) {
     return std::string(buffer.GetString());
 }
 
+future<> print(const rjson::value& value, seastar::output_stream<char>& os, size_t max_nested_level) {
```

Comment:
The implementation sort of assumes we're gonna be writing to the http output stream, yes, where we have ~32k of buffer to. Obviously, when _real_ large allocations might be in play, we probably generate more data than that, but the idea (if any) is to chain futures instead of building a buffer set, simply to _allow_ the stream to potentially push data to the actual output socket when buffers fill up. Of course, since we cannot yield in any meaningful way, chances are high we'll eventually get blocking futures that cannot resolve here. 

As a side note: To be able to handle these kinds of bridging cases, what we really need is a call `reactor::process_one_task`, so you could write `while (!f.available()) { reactor::process_one_task();  }` to essentially do message pumping on-stack until you can resume pushing data.